flink-core/src/main/java/org/apache/flink/util/AutoContextClassLoader.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.util;
+
+/**
+ * Sets a context class loader in a "try-with-resources" pattern.
+ *
+ * <pre>
+ * {@code
+ * try (AutoContextClassLoader ignored = AutoContextClassLoader.of(classloader)) {
+ *     // code that needs the context class loader
+ * }
+ * }
+ * </pre>
+ *
+ * <p>This is conceptually the same as the code below.
+
+ * <pre>
+ * {@code
+ * ClassLoader original = Thread.currentThread().getContextClassLoader();
+ * Thread.currentThread().setContextClassLoader(classloader);
+ * try {
+ *     // code that needs the context class loader
+ * } finally {
+ *     Thread.currentThread().setContextClassLoader(original);
+ * }
+ * }
+ * </pre>
+ */
+public final class AutoContextClassLoader implements AutoCloseable {
```

Comment:
duplicate of `TemporaryClassLoaderContext`

Reply:
Would consolidate that in a separate commit. If that gets used beyond plugins, it should reside in a more "universal" package. The `AutoContextClassLoader` is a tad bit nicer, especially javadoc wise.